### PR TITLE
Bump sklearn version

### PR DESCRIPTION
**Pull request by oleksandr-pavlyk**
Bumped up version of sklearn patches are known to be compatible with.

Updated version of Intel(R) DAAL for which certain examples (stump, boosting) apply.

Updated version of the packed in meta.yaml

Because DAAL is known to be non-deterministic in the constructed model for decision forest regression, exclude `check_fit_idempotent`  for `RandomForestRegressor` in  test_estimators.py

DAAL team has been notified.

@fschlimb @Alexander-Makaryev

### CODE REVIEW

File: examples/run_examples.py [hunk 1]
```diff
@@ -41,11 +41,11 @@
 req_version = defaultdict(lambda:(2019,0))
 req_version['decision_forest_classification_batch.py'] = (2019,1)
 req_version['decision_forest_regression_batch.py'] = (2019,1)
-req_version['adaboost_batch.py'] = (2019,4)
-req_version['brownboost_batch.py'] = (2019,4)
-req_version['logitboost_batch.py'] = (2019,4)
-req_version['stump_classification_batch.py'] = (2019,4)
-req_version['stump_regression_batch.py'] = (2019,4)
```
Comment (fschlimb) [inline examples/run_examples.py]: These version checks are done in a separate PR #99. The correct version would be 2020.1

Comment (oleksandr-pavlyk) [reply examples/run_examples.py]: So would you rather I merge #99 into this branch, and we merge these changes together, or I drop changes to these lines breaking CI and wait until #99 is merged ?

Comment (fschlimb) [reply examples/run_examples.py]: I suggest the latter, mergine #99 and then yours should work just fine even if you rebase your's onto #99 now

Comment (oleksandr-pavlyk) [reply examples/run_examples.py]: I have cherry-picked 2 commits changing this versions from the branch behind #99.

The change to `normalization_zscore` in #99 to use `doScale` is better in my PR, since it allows daal4py to build and work with older DAAL versions as well
